File: Easy/Java/Summary_Ranges(228).java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> result = new ArrayList();
        for(int i=0;i<nums.length;i++){
            int start = nums[i];
            while(i+1<nums.length && nums[i+1]== nums[i]+1){
                i++;
            }
            if(nums[i]==start){
                result.add(""+start);
            }
            else{
                result.add(""+start+"->"+nums[i]);
            }
        }
        return result;
    }
}

File: Easy/C++/Summary_Ranges(228).cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        
        for (int l = 0, r = 0; r < nums.size(); r++) {
            if (r == nums.size() - 1 || nums[r] + 1 < nums[r + 1]) {
                ans.push_back(to_string(nums[l]) + (l == r ? "" : "->" + to_string(nums[r])));
                l = r + 1;
            }
        }
        
        return ans;
    }
};